ml-agents/mlagents/envs/brain.py

```diff
@@ -18,14 +18,27 @@ def __init__(self, visual_observation, vector_observation, text_observations, me
         self.visual_observations = visual_observation
         self.vector_observations = vector_observation
         self.text_observations = text_observations
-        self.memories = memory
-        self.rewards = reward
-        self.local_done = local_done
-        self.max_reached = max_reached
-        self.agents = agents
-        self.previous_vector_actions = vector_action
-        self.previous_text_actions = text_action
-        self.action_masks = action_mask
+        self.memories = memory or [[]]
+        self.rewards = reward or []
+        self.local_done = local_done or []
+        self.max_reached = max_reached or []
+        self.agents = agents or []
+        self.previous_vector_actions = vector_action or [[]]
+        self.previous_text_actions = text_action or []
+        self.action_masks = action_mask or [[]]
+
+    def merge(self, other):
```

Comment:
Can you do something like `other: BrainInfo`?

Reply:
@eshvk that's not possible since it is happening within the definition of `BrainInfo`.